### PR TITLE
commitafter: fix ambiguous head error

### PR DESCRIPTION
**Pull request by kzh**

### CODE REVIEW

File: pkg/vcs/git/commits.go [hunk 1]
```diff
@@ -196,7 +205,7 @@ func commitLogArgs(initialArgs []string, opt CommitsOptions) (args []string, err
 	}
 
 	if opt.After != "" {
-		args = append(args, "--after="+opt.After)
+		args = append(args, "--after=\""+opt.After+"\"")
```
Comment (slimsag) [inline pkg/vcs/git/commits.go]: Why does this need to be quoted? I don't think it does need to be

File: pkg/vcs/git/commits.go [hunk 2]
```diff
@@ -219,10 +228,6 @@ func CommitCount(ctx context.Context, repo gitserver.Repo, opt CommitsOptions) (
 	span.SetTag("Opt", opt)
 	defer span.Finish()
 
-	if opt.Range == "" {
-		opt.Range = "HEAD"
-	}
```
Comment (slimsag) [inline pkg/vcs/git/commits.go]: Why is this changing? doesn't seem related

Comment (slimsag) [reply pkg/vcs/git/commits.go]: In particular, the reason I suggest not changing this is because there could be callers of this in the future that want to rely on this behavior

Comment (kzh) [reply pkg/vcs/git/commits.go]: I added this in my initial implementation of `repohascommitafter` so it doesn't seem necessary anymore with the addition of this PR. I was probably the only one making use of this.

File: pkg/vcs/git/commits.go [hunk 3]
```diff
@@ -112,9 +112,18 @@ func HasCommitAfter(ctx context.Context, repo gitserver.Repo, date string, revsp
 	span.SetTag("RevSpec", revspec)
 	defer span.Finish()
 
+	if revspec == "" {
+		revspec = "HEAD"
+	}
+
+	commitid, err := ResolveRevision(ctx, repo, nil, revspec, nil)
```
Comment (slimsag) [inline pkg/vcs/git/commits.go]:
```suggestion
	// We resolve the revision first because some repositories unfortunately have *both* a HEAD branch
	// and remotes/origin/HEAD -> master reference. We've only encountered this on k8s.sgdev.org (see
	// https://sourcegraph.slack.com/archives/CHEKCRWKV/p1561004415100200?thread_ts=1560992541.075000&cid=CHEKCRWKV) and CommitCount below would fail to take into
	// account the first line that is output:
	//
	//  warning: refname 'HEAD' is ambiguous.
	// 
	// The reason we can't just solve this in CommitCount by handling the above output is because...
	commitid, err := ResolveRevision(ctx, repo, nil, revspec, nil)
```

Someone may look at this code in the future and think we can simplify it to just pass `revspec` into `CommitCount` below, so we should add a comment here.

File: pkg/vcs/git/commits.go (hunk 2 above)
Comment (slimsag) [reply pkg/vcs/git/commits.go]: OK, removing LGTM then.

File: pkg/vcs/git/commits.go [hunk 4]
```diff
@@ -112,9 +112,36 @@ func HasCommitAfter(ctx context.Context, repo gitserver.Repo, date string, revsp
 	span.SetTag("RevSpec", revspec)
 	defer span.Finish()
 
+	if revspec == "" {
+		revspec = "HEAD"
+	}
+
+	// We resolve the revision first because some repositories unfortunately have *both* a HEAD branch
+	// and remotes/origin/HEAD -> master reference. We've only encountered this on k8s.sgdev.org (see
+	// https://sourcegraph.slack.com/archives/CHEKCRWKV/p1561004415100200?thread_ts=1560992541.075000&cid=CHEKCRWKV)
```
Comment (slimsag) [inline pkg/vcs/git/commits.go]:
```suggestion
	// https://github.com/sourcegraph/sourcegraph/issues/4619)
```

File: pkg/vcs/git/commits.go [hunk 5]
```diff
@@ -112,9 +112,36 @@ func HasCommitAfter(ctx context.Context, repo gitserver.Repo, date string, revsp
...
+		revspec = "HEAD"
+	}
+
+	// We resolve the revision first because some repositories unfortunately have *both* a HEAD branch
+	// and remotes/origin/HEAD -> master reference. We've only encountered this on k8s.sgdev.org (see
+	// https://github.com/sourcegraph/sourcegraph/issues/4619)
+	// and CommitCount below would fail to take into account the first line that is output:
+	//
+	//  warning: refname 'HEAD' is ambiguous.
+	//
```
Comment (slimsag) [inline pkg/vcs/git/commits.go]:
```suggestion
	//
```

File: pkg/vcs/git/commits.go [hunk 6]
```diff
@@ -112,9 +112,36 @@ func HasCommitAfter(ctx context.Context, repo gitserver.Repo, date string, revsp
...
+	commitid, err := ResolveRevision(ctx, repo, nil, revspec, nil)
+	if err != nil {
+		return false, err
+	}
+
 	n, err := CommitCount(ctx, repo, CommitsOptions{
 		After: date,
-		Range: revspec,
+		Range: string(commitid),
```
Comment (keegancsmith) [inline pkg/vcs/git/commits.go]: @slimsag @kzh is this change correct? We go from supporting a rev range to requiring a specific rev. I think we should revert. ie before `revspec` could be `featurebranch..master`, that won't work now.

Comment (keegancsmith) [reply pkg/vcs/git/commits.go]: Oh I see this is just for commitafter, so this is actually fine. Nevermind me :)
